### PR TITLE
cleanup + unit test for --cleanup-tmpdir

### PR DESCRIPTION
**Pull request by boegel**
for https://github.com/hpcugent/easybuild-framework/pull/1365

### CODE REVIEW

File: easybuild/tools/filetools.py [hunk 1]
```diff
@@ -929,25 +929,26 @@ def move_logs(src_logfile, target_logfile):
...
-        try:
-            for log in glob.glob('%s*' % logfile):
-                os.remove(log)
-        except OSError, err:
-            raise EasyBuildError("Failed to remove log file(s) %s*: %s", logfile, err)
-        print_msg('temporary log file(s) %s* have been removed.' % (logfile), log=None, silent=testing)
+    if build_option('cleanup_tmpdir') and not testing:
+        if logfile is not None:
+            try:
+                for log in glob.glob('%s*' % logfile):
```
Comment (wpoely86) [inline easybuild/tools/filetools.py]: why the wildcard?

Comment (boegel) [inline easybuild/tools/filetools.py]: log files can be rotated: `*log.1`, etc.

Comment (wpoely86) [inline easybuild/tools/filetools.py]: okay, be shouldn't we use that specific structure? I'm not sure I'm comfortable with a general wildcard here

Comment (boegel) [inline easybuild/tools/filetools.py]: WHat do you mean? Only consider `.log.0` & friends?

Comment (wpoely86) [inline easybuild/tools/filetools.py]: yes

Comment (boegel) [inline easybuild/tools/filetools.py]: ok, fixed, now only `*.log`, `*.log.0`, etc will be removed

**Pull request merged**